app/controllers/screenings_controller.rb

```diff
@@ -0,0 +1,58 @@
+class ScreeningsController < ApplicationController
+  before_action :set_screening, only: %i[show edit update destroy]
+  before_action :set_hall, :set_movie, only: %i[new create edit update]
+
+  def index
+    @screenings = Screening.all
+  end
+
+  def new
+    @screening = Screening.new
+  end
+
+  def create
+    @screening = Screening.new(screening_params)
+
+    if @screening.save
+      redirect_to screenings_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    if @screening.update(screening_params)
+      redirect_to @screening
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @screening.destroy
+
+    redirect_to screenings_path, status: :see_other
+  end
+
+  private
+
+  def set_screening
+    @screening = Screening.find(params[:id])
+  end
+
+  def set_hall
+    @halls = Hall.all.order(:name)
+  end
+
+  def set_movie
+    @movies = Movie.all.order(:title)
+  end
```

Comment:
singular form in those names suggest you are setting one hall and one movie
these should be called set_halls and set_movies to indicate you are setting lists of objects

Reply:
good idea, thank you